packages/profile-sync-controller/src/controllers/user-storage/UserStorageController.test.ts

```diff
@@ -444,20 +1290,40 @@ function mockUserStorageMessenger() {
       NotificationServicesControllerDisableNotificationServices['handler']
     >().mockResolvedValue();
 
+  const mockKeyringAddNewAccount = jest.fn().mockResolvedValue('0x123');
+
+  const mockAccountsUpdateAccountMetadata = jest.fn().mockResolvedValue(true);
+
+  const mockAccountsGetAccountByAddress = jest.fn().mockResolvedValue({
+    address: '0x123',
+    id: '1',
+    metadata: {
+      name: 'test',
+      nameLastUpdatedAt: 1,
+    },
+  });
+
   jest.spyOn(messenger, 'call').mockImplementation((...args) => {
     const [actionType, params] = args;
     if (actionType === 'SnapController:handleRequest') {
-      if (params?.request.method === 'getPublicKey') {
+      if (
+        (params as Exclude<typeof params, string | number>)?.request.method ===
```

Comment:
yep get it, the types for this mock are a little annoying. Will investigate.

Reply:
Yep, `messenger.call` unions the actions and parameters which makes it hard grab the right type.

Use this to shim in the correct types.

```ts
  // Creates the correct typed call params for mocks
  type CallParams = {
    [K in AllowedActions['type']]: [
      K,
      ...Parameters<Extract<AllowedActions, { type: K }>['handler']>,
    ];
  }[AllowedActions['type']];
  
  ...
  const [actionType, Params] = args as unknown as CallParams;
```

Reply:
***Correction

The underlying `call` method is typed correctly, but I'm suspecting the `jest.spyOn` is not playing well with generic type parameters and is "expanding" the type, hence the CallParams are a union.

Reply:
Nice one @Prithpal-Sooriya , added!